src/Business/Provider/FactoryProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Micro\Plugin\Serializer\Business\Provider;
+
+use Micro\Framework\Kernel\KernelInterface;
+use Micro\Plugin\Serializer\Business\Factory\SerializerFactoryInterface;
+use Micro\Plugin\Serializer\Exception\SerializerFactoryNotFoundException;
+use Micro\Plugin\Serializer\Plugin\SerializerFactoryProviderPluginInterface;
+
+readonly class FactoryProvider implements FactoryProviderInterface
+{
+    public function __construct(private KernelInterface $kernel)
+    {
+    }
+
+    public function getFactory(): SerializerFactoryInterface
+    {
+        $iterator = $this->kernel->plugins(SerializerFactoryProviderPluginInterface::class);
+
+        /** @var SerializerFactoryProviderPluginInterface $serializerFactoryProviderPlugin */
+        foreach ($iterator as $serializerFactoryProviderPlugin) {
+            return $serializerFactoryProviderPlugin->provideSerializerFactory();
```

Comment:
You really wants to return first available plugin ?

Reply:
MR updated